lib/DataDictionary.php

```diff
@@ -0,0 +1,22 @@
+<?php
+namespace Octri\ConsortReport;
+
+/**
+ * A class for working with a project's data dictionary.
+ */
+class DataDictionary {
+
+    private $dictionary;
+
+    /**
+     * @param Array dictionary Data dictionary as returned by `REDCap::getDataDictionary('array')`
+     */
+    public function __construct($dictionary) {
+        $this->dictionary = $dictionary;
+    }
+
+    public function getFieldLabel($fieldName) {
+        return $this->dictionary[$fieldName]['field_label'];
```

Comment:
If you are switching from calling REDCap's API that returns json to its API that returns an array, I'd just be really careful to ensure you inspect what you're getting back. I remember having some issues when I worked on VIZR where only one or the other of the methods actually returned everything I needed. They are not equivalent. I think it was the array method that was more complete.

Reply:
I just doubled checked and I don't see any differences in the code other than building json. This is how it handles json vs array.
```
// Change structure according to desired reteurn format^M
        switch ($returnFormat)^M
        {^M
            // Array (PHP array type)^M
            case 'array':^M
                return $lines;^M
                break;^M
            // JSON^M
            case 'json':^M
                // Convert all data into JSON string (do line by line to preserve memory better)^M
                $content = '';^M
                foreach ($lines as $key=>&$item) {^M
                    // Loop through each record and encode^M
                    $item_json = json_encode_rc($item);^M
                    if ($item_json !== false) $content .= ",\n".$item_json;^M
                    // Remove line from array to free up memory as we go^M
                    unset($lines[$key]);^M
                }^M
                return '[' . substr($content, 2) . ']';^M
                break;^M
```

Reply:
This may be okay because it's the data dictionary, but other REDCap calls will return different structures depending on the output type. IIRC, record data is structured very differently in the two formats, particularly with  repeating instruments.

Reply:
That makes sense - I remember hearing about those issues and wondered if it might be the `getData` call.